File: Java/NetBeans2/QLNV/src/qlnv/NV.java
package qlnv;

import java.io.Serializable;

public abstract class NV implements Serializable {

    protected int maNV;
    protected String hoTen;
    protected int namSinh;
    protected int soNamCT;
    protected float luongCB;

    public NV(int maNV, String hoTen, int namSinh, int soNamCT, float luongCB) {
        this.maNV = maNV;
        this.hoTen = hoTen;
        this.namSinh = namSinh;
        this.soNamCT = soNamCT;
        this.luongCB = luongCB;
    }

    public float layLuongCB() {
        return luongCB;
    }

    public void ganLuongCB(float luongCB) {
        this.luongCB = luongCB;
    }

    public abstract float tinhLuong();

    public abstract String loaiNV();

    public String toString() {
        return maNV + ";" + hoTen + ";" + namSinh + ";" + soNamCT + ";" + luongCB;
    }

    public boolean equals(Object obj) {
        if (obj instanceof NV) {
            NV nv = (NV) obj;
            if (maNV == nv.maNV && hoTen.equals(nv.hoTen) && namSinh == nv.namSinh && soNamCT == nv.soNamCT && luongCB == nv.luongCB) {
                return true;
            }
        }

        return false;
    }
}

File: C#/giuaKY/giuaKY/NV.cs
namespace giuaKY
{
    abstract public class NV
    {
        protected int maNV;
        protected string hoTen;
        protected int namSinh;
        protected int soNamCT;
        protected float luongCB;

        public NV(int maNV, string hoTen, int namSinh, int soNamCT, float luongCB)
        {
            this.maNV = maNV;
            this.hoTen = hoTen;
            this.namSinh = namSinh;
            this.soNamCT = soNamCT;
            this.luongCB = luongCB;
        }

        public virtual void nhap()
        {
            Console.Write("Nhap ma nhan vien: ");
            maNV = Convert.ToInt32(Console.ReadLine());
            Console.Write("Nhap ho ten: ");
            hoTen = Console.ReadLine();
            Console.Write("Nhap nam sinh: ");
            namSinh = Convert.ToInt32(Console.ReadLine());
            Console.Write("Nhap so nam cong tac: ");
            soNamCT = Convert.ToInt32(Console.ReadLine());
            Console.Write("Nhap luong co ban: ");
            luongCB = Convert.ToSingle(Console.ReadLine());
        }

        abstract public float tinhLuong();
        abstract public string loaiNV();
        public override string ToString()
        {
            return maNV + ";" + hoTen + ";" + namSinh + ";" + soNamCT + ";" + luongCB;
        }
    }
}
